src/shims-vue-router.d.ts

```diff
@@ -6,5 +6,10 @@ declare module 'vue-router' {
      * The module this route belongs to
      */
     module?: string
+
+    /**
+     * Whether to ignore this route in the forming of navigational tabs
+     */
+    shouldIgnoreInNavTabs?: boolean
```

Comment:
How come this is needed?

Reply:
Did I understand right that we would have a nested routing structure here once we remove the feature flag? i.e.

- `/mesh/default/services/`
  - `/mesh/default/services/internal`
  - `/mesh/default/services/external`

If so that would mean that this ceases to be something we need to work around because we are using nested routing. That being said can we just filter out the route with the name `external-service-list-view`. FYI I don't want this sort of `route.meta` to become a thing. If for some reason this sort of workaround is needed, its more likely that the code is going against the natural nested routing approach, and its more important to stick with that nested routing approach rather than workaround things.

Lastly I get why you are trying to not alter the URL structure too much for the feature flag reasons, but this is also why I think its fine to just use the route name in the filter instead of adding new `route.meta` i.e. its a temporary thing.

Reply:
> Did I understand right that we would have a nested routing structure here once we remove the feature flag? i.e.
> 
> * `/mesh/default/services/`
>   
>   * `/mesh/default/services/internal`
>   * `/mesh/default/services/external`

I haven’t made up my mind on this, yet. _Internal_ services aren’t really a thing. We have ServiceInsights which represent all manner of service in Kuma which is confusing because those services aren’t very similar. We might get a dedicated resource in MeshService that will replace ServiceInsight, but that’s not a concrete plan, yet. The term "Internal" kind of just appears because we have ExternalServices. Long story, short: I want to avoid the notion that we have two types of services that are just slight variations.

To that end, I prefer the routes using `services/` (which in the future could be replaced by `mesh-services/`) and `external-services/`.

> If so that would mean that this ceases to be something we need to work around because we are using nested routing. That being said, is it possible to just filter out the route with the name `external-service-list-view`? FYI I don't want this sort of `route.meta` to become a thing. If for some reason this sort of workaround is needed, its more likely that the code is going against the natural nested routing approach, and its more important to stick with that nested routing approach rather than workaround things.
> 
> Lastly I get why you are trying to not alter the URL structure too much for the feature flag reasons, but this is also why I think its fine to just use the route name in the filter instead of adding new `route.meta` i.e. its a temporary thing.

Not sure what’s bad using `route.meta` as a place to store such filter decisions. It’s functionally the same as filtering the route by name but simpler.

It’s true that nesting both types of services under `services/` would allow us to avoid having to filter the external service route from participating in nav tabs generation, but before I know what URL paths we’ll settle on, I’m not going to re-write the route setup for this PR (again) just so I can avoid the small additions to `route.meta`.

Reply:
> Is there any other reason to feature flag this "services to internal/external services" feature?

The other reason was that this set of features will take multiple PRs to land. We want to minimize the amount of UI reorganization changes.

The feature flag will remain in place during the development here even if the necessary API capabilities are available.

Reply:
Ok so if the only potential reason not to do nested routing here is possibly due to the feature flagging, will this be moved to use nested routing once we remove the feature flag?

Reply:
Possibly. I’m not making a decision on this now.